atsign-foundation/at_libraries
# fix: Add unrevoke to the at_onboarding_cli

Closes #627 and #628

This PR includes everything from #670 

**- What I did**
- Add "unrevoke" to auth_cli commands which will restore the revoked enrollmentId.
- Add "delete" to auth_cli commands which will permanently delete enrollment requests (only denied/revoked enrollments may be deleted)
- Made various changes (replacing `exit` with `throw`, etc) to ensure that the tests in enrollment_cli_commands_test.dart behave as they should

**- How I did it**
- Introduced 'delete' as an option in the at_activate argParser
- Introduced a delete-command parser which considers enrollmentId mandatory
- Introduce a new method deleteDeniedEnrollment which sends a delete request to the server and parses the server response.
- Add "unrevoke" command to auth_cli_args list.
- Add "unrevoke" method in auth_cli which will do the necessary work to unrevoke the enrollment Id.

**- How to verify it**
- Added an End-to-End test in "enrollment_cli_commands_test.dart" file which covers unrevoke functionality.
- Manual testing has been done. 
- To test the changes, send an enrollment request then deny/approve+revoke it. Then run at_activate with the delete option which requires the atsign and enrollmentId. The enrollment should be deleted.
